src/DynamoCore/Configuration/PreferenceSettings.cs

```diff
@@ -828,21 +825,50 @@ internal void SetTrustWarningsDisabled(bool disabled)
             disableTrustWarnings = disabled;
         }
 
+        // Add default trusted locations for Autodesk samples.
+        // This function should only be called during Dynamo's first run.
+        internal void AddDefaultTrustedLocations()
+        {
+            if (!IsFirstRun) return;
+
+            const string Autodesk = "Autodesk";
+            string ProgramData = Environment.GetFolderPath(Environment.SpecialFolder.CommonApplicationData);
+            AddTrustedLocation(Path.Combine(ProgramData, Autodesk));
+
+            string ProgramFiles = Environment.GetFolderPath(Environment.SpecialFolder.ProgramFiles);
+            AddTrustedLocation(Path.Combine(ProgramData, Autodesk));
+        }
+
+        /// <summary>
+        /// Returns true if the input "location" is among the stored trusted paths.
+        /// Subdirectories of a trusted path are considered trusted.
+        /// Does not validate the input for correctness.
+        /// </summary>
+        /// <param name="location"></param>
+        /// <returns></returns>
+        private bool _isTrustedLocation(string location)
```

Comment:
I think the coding convention is always to capitalize function names.

Reply:
isTrustedLocationInternal